src/store/search-store.js

```diff
@@ -166,10 +165,25 @@ const actions = ImageService => ({
         );
       })
       .catch((error) => {
-        commit(FETCH_IMAGES_ERROR);
-        throw new Error(error);
+        dispatch(HANDLE_IMAGE_ERROR, error);
       });
   },
+  [HANDLE_IMAGE_ERROR]({ commit }, error) {
+    if (error.response) {
+      if (error.response.status === 500) {
+        commit(FETCH_IMAGES_ERROR, { errorMsg: 'There was a problem with our servers' });
+      }
+      else if (error.response.status === 404) {
+        commit(FETCH_IMAGES_ERROR, { errorMsg: 'No images were found for this image' });
+      }
+      else {
+        commit(FETCH_IMAGES_ERROR, { errorMsg: error.response.message });
+      }
+      return;
+    }
+    commit(FETCH_IMAGES_ERROR, { errorMsg: error.message });
```

Comment:
Can you put an `else` here, otherwise you will be committing `FETCH_IMAGES_ERROR` twice.